sdk/nodejs/provider/server.ts

```diff
@@ -708,7 +708,16 @@ function grpcResponseFromError(e: { id: string; properties: any; message: string
     };
 }
 
+// Add a flag to track if main has already been called
+let isServing = false;
```

Comment:
Do we need this check here? I think the guard in `componentProviderHost` is sufficient.

Reply:
I added it in case a user would call the server directly, bypassing componentProviderHost

Reply:
> I added it in case a user would call the server directly, bypassing componentProviderHost

👍🏻 I was actually experimenting with doing this